### DISCUSSION TITLE
Should I not use useSession hook and SessionProvider on V5 with App Router?

### DISCUSSION THREAD

**Discussion by dBianchii** [nextauthjs/next-auth#9137] [Help]
Hi everyone. Recently I noticed that in V5, the SesionProvider tooltip says we shouldn't / don't need to use it on the app router. This made me somewhat confused because in the docs, they say that the way to access the session on client components was to use the `useSession` hook. After some research, I found that some people were saying that the way to do it in V5 is to call `auth()` on the server component, and pass it down to client if it needs it. But I am unsure. Is it correct to do it like that?

I completely removed useSession from my app. In this PR:
https://github.com/dBianchii/kodix-turbo/pull/40 

So, what is the recomended way? Does the useSession support only exist now for `/pages` ?

**Comment by manuelatdev** [comment]
I have the same question, it seems that the recommended way is always to get the session via auth() on the server, at page level, and then pass it to the components, but that can be very tedious because of the prop drilling. Are we supposed to use something like zustand and keep the session there to provide global access?

**Reply by Dragate** [reply]
No. You can still use `SessionProvider` if you insist on using the `useSession` hook.

On pages router, you would need to do a lot of prop-drilling to pass the session to underlying components. `SessionProvider` existed to address that issue. Now on app router, every server component can access the session with no prop-drilling via the `auth`  function. Even if your component is a client component, it most likely has a server component as parent and can receive the session from them directly. This makes fetching/providing the session on the client through context almost redundant.

**Reply by Jaycoda** [reply]
so SessionProvider is no longer neccessary to use in V5 using the app router?
